src/psyclone/psyir/symbols/symbol.py

```diff
@@ -233,13 +232,29 @@ def __init__(self, name, visibility=DEFAULT_VISIBILITY, interface=None):
         # The following attributes have a setter method (with error checking)
         self._visibility = None
         self._interface = None
-        # If an interface is not provided, use LocalInterface by default
-        if not interface:
-            self._interface = LocalInterface()
-        else:
-            # Use the setter as it checks the variables validity
+
+        self._process_arguments(visibility=visibility, interface=interface)
+
+    def _process_arguments(self, visibility=None, interface=None):
+        ''' Process the visibility and interface arguments of the constructor
+        and the specialise methods.
+
+        :type visibility: :py:class:`psyclone.psyir.symbols.Symbol.Visibility`
+        :param interface: optional object describing the interface to this \
+            symbol (i.e. whether it is passed as a routine argument or \
+            accessed in some other way). Defaults to \
+            :py:class:`psyclone.psyir.symbols.LocalInterface`
+        :type interface: \
+            :py:class:`psyclone.psyir.symbols.symbol.SymbolInterface`
+
+        '''
+        if interface:
             self.interface = interface
-        self.visibility = visibility
+        elif not self.interface:
+            self.interface = LocalInterface()
+
+        if visibility:
+            self.visibility = visibility
```

Comment:
I don't think it can ever happen as any starting symbol will call to the constructor above and already have a DEFAULT_VISIBILITY if not specified.

I thought of moving the default value down to the _process_arguments to make it more consistent with the interface attribute but then it doesn't appear in the generated documentation of the class. So I decided to keep it in the constructor.